AzureCalling/Controllers/CallViewController.swift

```diff
@@ -462,37 +462,76 @@ class CallViewController: UIViewController, UICollectionViewDelegate, UICollecti
         var indexPathMoves: [(at: IndexPath, to: IndexPath)] = []
         var insertIndexPaths: [IndexPath] = []
 
-        // Build new maps and collect changes
-        for (index, participant) in callingContext.displayedRemoteParticipants.enumerated() {
-            let userIdentifier = participant.identifier.stringValue ?? ""
-            let indexPath = IndexPath(item: index, section: 0)
-            var participantView: ParticipantView
-
-            // Check for previously tracked participants
-            if let prevIndexPath = prevParticipantIdIndexPathMap[userIdentifier],
-               let prevParticipantView = prevParticipantIndexPathViewMap[prevIndexPath] {
-                prevParticipantIndexPathViewMap.removeValue(forKey: prevIndexPath)
+        // Check to see if any screenshare
+        var screenShareParticipantView: ParticipantView?
+        outerLoop: for participant in callingContext.displayedRemoteParticipants {
+            for videoStream in participant.videoStreams {
+                if videoStream.mediaStreamType == .screenSharing {
```

Comment:
The following part of code is kind of redundant with the line 504-533, could it be possible to abstract one function with different parameters in put to build the participant view?

Reply:
The logic is now in callingContext